VoxelSniperCore/src/main/java/com/github/kevindagame/brush/multiBlock/GenerateTreeBrush.java

```diff
@@ -121,10 +121,29 @@ private void leafNodeCreate(final BaseLocation origin) {
     private void createLeaf(final BaseLocation location, int x, int y, int z) {
         if (location.getBlock().getRelative(x, y, z).getMaterial().isAir()) {
             var block = location.getBlock().getRelative(x, y, z);
-            addOperation(new BlockOperation(block.getLocation(), block.getBlockData(), this.leavesMaterial.createBlockData()));
+            var blockData = this.leavesMaterial.createBlockData();
+            if (blockData instanceof IPersistent) {
+                ((IPersistent) blockData).setPersistent(true);
+            }
+
+            var newLocation = block.getLocation();
+            if (leafBlocks.get(getKey(newLocation)) == null) {
+                addOperation(new BlockOperation(newLocation, block.getBlockData(), blockData));
+                leafBlocks.put(getKey(newLocation), true);
+            }
         }
     }
 
+    /**
+     * Generate a key for a location
+     *
+     * @param location the location
+     * @return "x:y:z"
+     */
+    private String getKey(final BaseLocation location) {
+        return location.getX() + ":" + location.getY() + ":" + location.getZ();
```

Comment:
Should we use getBlock[XYZ] here? do we even need to make it a string?

Reply:
The string is to feed the hashSet. Making it a string is cheaper than passing the whole object since it should be unique by location rather than by object.

Reply:
a location with the same coords is a different object, so passing a whole location wouldn't work

Reply:
location has an equals and hashcode, but string is also fine